kani-compiler/src/codegen_cprover_gotoc/overrides/hooks.rs

```diff
@@ -316,6 +417,49 @@ impl<'tcx> GotocHook<'tcx> for PtrWrite {
     }
 }
 
+/// Hook to set flag to nondet the function body if a function is being replaced by its contract.
+struct ReplaceFunctionBody;
+
+impl<'tcx> GotocHook<'tcx> for ReplaceFunctionBody {
+    fn hook_applies(&self, tcx: TyCtxt<'tcx>, instance: Instance<'tcx>) -> bool {
+        matches_function(tcx, instance, "KaniReplaceFunctionBody")
```

Comment:
I am not sure I understand how this hook would work. Who is calling this function? Note that the hooks get applied at a function call level not function body code generation.

Reply:
The macro expansion stage would call this function. The code would look something like this - 
```
if kani::replace_function_body() {
  let ret = kani::any();
} else {
  let ret = .... //call body closure.
}
```